feat: Add SwiftDescriptor error fix to troubleshooting for RN and Apple (#10087)
getsentry/sentry-docs: docs/platforms/apple/common/troubleshooting/index.mdx

Since Xcode 14 App Store Connect doesn't make debug symbols available for download anymore, see [Xcode Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes#Deprecations). Please use a different way of uploading the debug symbols by following the [documentation](/platforms/apple/dsym/).

## Unknown receiver 'SwiftDescriptor'

> If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, it's due to a conflict of `APPLICATION_EXTENSION_API_ONLY`. The Sentry cocoapod is by default configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script changed it. To fix the error add the following to the `Podfile`.

Reviewer: `l`: Minor fixes
```suggestion
If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, it's due to a conflict of `APPLICATION_EXTENSION_API_ONLY`. By default, the Sentry CocoaPod is configured with `APPLICATION_EXTENSION_API_ONLY=YES` but some `post_install` script changed it. To fix the error, add the following to the `Podfile`.
```

Reviewer 2: ```suggestion
If you see an error like `Unknown receiver 'SwiftDescriptor'` during the application build process, there's a conflict stemming from `APPLICATION_EXTENSION_API_ONLY`. By default, the Sentry Cocoapod is configured with `APPLICATION_EXTENSION_API_ONLY=YES`, but some `post_install` script likely changed it. To fix this error, add the following to the `Podfile`.
```
